Linstor support

ydirson opened the pull request.

ydirson committed:
Use skip_missing_names_on_install=0 in yumdl templates too

This default value is so nonsensical, better safe than sorry.

Signed-off-by: Yann Dirson <yann.dirson@vates.tech>

ydirson committed:
create-iso: new --extra-packages option

Meant to be used for generation of images with a repo including linstor
packages.

Signed-off-by: Yann Dirson <yann.dirson@vates.tech>

ydirson committed:
linstor: support for including RPMs in repository

This adds:
- adjustment to mirror-repos to stop excluding Java packages, some being
  required by LINSTOR
- addition of a `linstor` layer to point to the repository containing
  the LINSTOR packages

Signed-off-by: Yann Dirson <yann.dirson@vates.tech>

stormi commented:
LGTM in the current WIP stage.

stormi commented on README.md:
```diff
@@ -205,6 +205,44 @@ sudo ./scripts/create-installimg.sh \
+    --output xcp-ng-8.3.linstor.iso \
+    --extra-packages "xcp-ng-linstor" \
+    -V "XCP-NG_830_TEST" \
+    8.3:testing:linstor install-8.3.img
```
Why `testing` here as the previous command uses `updates`?

ydirson replied:
fixed

ydirson commented on scripts/create-iso.sh:
```diff
@@ -231,7 +239,7 @@ if [ $DOREPO = 1 ]; then
 
     mkdir ${VERBOSE} "$ISODIR/Packages"
 
-    get_rpms --depends "$ISODIR/Packages" xcp-ng-deps kernel-alt
+    get_rpms --depends "$ISODIR/Packages" xcp-ng-deps kernel-alt ${EXTRA_PACKAGES}
```
After stepping back, this clearly calls for getting packages from template files directly, the hardcoded `xcp-ng-deps kernel-alt` should have inspired this to start with - but given the timeframe, that will be for later (and that new option can still be useful)

ydirson commented on configs/linstor/yum-repos.conf.tmpl:
```diff
@@ -0,0 +1,6 @@
+baseurl=https://repo.vates.tech/xcp-ng/8/8.3/linstor/@@RPMARCH@@/
+gpgkey=https://xcp-ng.org/RPM-GPG-KEY-xcpng
+failovermethod=priority
+skip_if_unavailable=False
```
That last one is the default documented value already, we can surely drop it (and from everywhere else).

~~OTOH it is missing `skip_missing_names_on_install=0`~~ Edit: in fact no, that one is a `[main]` flag, not a per-repo one

ydirson committed:
Make SRCURL a part of the lib

SRCURL existence is set by `maybe_set_srcurl`, and subsequently assumed
present by `setup_yum_repos`, it makes no sense to give it to
`setup_yum_download` which just "sets it again".

We're going to introduce the SRCURLS[] array which has a similar
status, so better set things straight for its friend first.

Signed-off-by: Yann Dirson <yann.dirson@vates.tech>

ydirson committed:
Improve overlay documentation

`yum-repos.conf.tmpl` was the only file not mentionned by name, make the
description of repo layers consistent with that of version layers.

Also clarify that `base` *has* to be explicitly in the include chain.

Signed-off-by: Yann Dirson <yann.dirson@vates.tech>

ydirson committed:
create-iso: add --srcurl:<overlay> support

This is designed to allow mirroring non-official repositories, notably
the linstor one.

Signed-off-by: Yann Dirson <yann.dirson@vates.tech>

glehmann reviewed (approved):
I've checked that the built image contains the linstor-* packages, but not tried installing it yet

stormi commented on README.md:
````diff
@@ -212,6 +212,36 @@ sudo ./scripts/create-installimg.sh \
+    8.3:updates:linstor install-8.3.img
+```
+
+> [!NOTE]
+>
+> This example only pulls the latest version of the LINSTOR packages.
+> To be suitable for update of a LINSTOR-enabled XCP-ng 8.2.1, it must
+> *also* be provided with the version matching the 8.2.1 installation.
````
If that's not handled in this PR, we'll need to spawn a new work item to address that before 8.2.1 EOL.

ydirson commented:
FIxed a typo in commit message

stormi reviewed (approved):
I built images with the updated process, nearly identical result to my previous batch (with the exception of host-installer having pulled the missing yum-utils RPM inside install.img as expected).
